db.go

```diff
@@ -128,6 +128,106 @@ func SmartUnmarshal[I any](respond interface{}, wrapperError error) (data I, err
 	return data, err
 }
 
+// SmartUnmarshalAll takes an interface input and unmarshals it into the given
+// type, traversing the nested slices and maps, regardless of the input being
+// wrapped in RawQuery or not. When any error is seen during the unmarshaling,
+// it would keep the return value along with the error message.
+func SmartUnmarshalAll[I any](input interface{}) ([]I, error) {
+	// NOTE: We could use a dedicated error type to provide unmarshal failure,
+	// which may allow users to ignore some erroneous data being returned from
+	// the database.
+	return handleInterfaces[I](input)
+}
+
+func handleInterfaces[I any](input interface{}) ([]I, error) {
+	var result []I
+	var errs []error
+	inputs, ok := input.([]interface{})
+	if !ok {
+		return handleInput[I](input)
+	}
+
+	for _, i := range inputs {
+		data, err := handleInterfaces[I](i)
+		errs = append(errs, err)
+		result = append(result, data...)
+	}
+	if len(errs) > 0 {
+		return result, errors.Join(errs...)
+	}
+	return result, nil
+}
+
+var errNotRawQuery = fmt.Errorf("not a RawQuery")
```

Comment:
This error shouldn't need to be public, but as mentioned in my comment above, it would probably make sense to add more errors so that package user can decide which error to ignore / handle.